gobblin-runtime/src/main/java/org/apache/gobblin/runtime/Task.java

```diff
@@ -173,7 +169,30 @@ public Task(TaskContext context, TaskStateTracker taskStateTracker, TaskExecutor
     this.extractor =
         closer.register(new InstrumentedExtractorDecorator<>(this.taskState, this.taskContext.getExtractor()));
 
-    this.converter = closer.register(new MultiConverter(this.taskContext.getConverters()));
+    this.recordStreamProcessors = this.taskContext.getRecordStreamProcessors();
+
+    // add record stream processors to closer if they are closeable
+    for (RecordStreamProcessor r: recordStreamProcessors) {
+      if (r instanceof Closeable) {
+        this.closer.register((Closeable)r);
+      }
+    }
+
+    List<Converter<?,?,?,?>> converters = this.taskContext.getConverters();
+
+    this.converter = closer.register(new MultiConverter(converters));
+
+    // can't have both record stream processors and converter lists configured
+    if (!this.recordStreamProcessors.isEmpty() && !converters.isEmpty()) {
```

Comment:
Just to make it a bit clearer, can you do something like `Preconditions.checkState(!this.recordStreamProcessor.isEmpty() ^ !this.converters.isEmpty(), "Mesasge...");`

Reply:
I need to call the closer here and both empty is valid so ^ won't work.

Reply:
Added precondition check wrapped by a try catch block.